be/src/exprs/array_functions.cpp

```diff
@@ -752,6 +752,73 @@ class ArrayHasImpl {
             return true;
         }
     }
+
+    template <bool NullableElement, bool NullableTarget, typename ElementColumn>
+    static uint8 __process_seq(const ElementColumn& elements, uint32 element_start, uint32 element_end,
+                               const ElementColumn& targets, uint32 target_start, uint32 target_end,
+                               const NullColumn::Container* null_map_elements,
+                               const NullColumn::Container* null_map_targets) {
+        using ValueType = std::conditional_t<std::is_same_v<ArrayColumn, ElementColumn> ||
+                                                     std::is_same_v<MapColumn, ElementColumn> ||
+                                                     std::is_same_v<StructColumn, ElementColumn>,
+                                             uint8_t, typename ElementColumn::ValueType>;
+        [[maybe_unused]] auto is_null = [](const NullColumn::Container* null_map, size_t idx) -> bool {
+            return (*null_map)[idx] != 0;
+        };
+        if (element_end - element_start < target_end - target_start) {
+            return false;
+        }
+        if (target_end == target_start) {
+            return true;
+        }
+        if (element_end == element_start) {
+            return false;
+        }
+        bool found = false;
+        size_t i = target_start;
+        size_t j = element_start;
+        while (j < element_end) {
```

Comment:
j <= element_end - (target_end - target_start), no need to loop if the left elements are less than target's elements